### PR TITLE
Notification Extensions: Adds currency symbol to order

### PR DESCRIPTION
**Pull request by Ecarrion**
# Why

For my HACK Week project, I'm working on providing a richer UI for order notifications. In order to accomplish this well, this PR adds the `currency_symbol` property to the order entity.

## Notes

- As the currency symbol is not used on the app, but only on the extension, the property is not stored on the storage layer. We can store it later if needed.

- After running rake generate, some changes were made to a Tax object, which forced me to update some tests.

# Testing Steps

- Make sure the test pass and orders can be loaded on the app.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerceTests/Model/AddressWooTests.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ class AddressWooTests: XCTestCase {
 
     func test_from_taxRate_when_tax_rate_has_location_then_prioritizes_array_properties() {
         // Given
-        let taxRate = TaxRate.fake().copy(country: "US", state: "CA", postcode: "12345", postcodes: ["9999"], city: "Miami", cities: ["New York"])
+        let taxRate = TaxRate.fake().copy(state: "CA", country: "US", postcode: "12345", postcodes: ["9999"], city: "Miami", cities: ["New York"])
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/Model/AddressWooTests.swift]: [these `TaxRate` changes](https://github.com/woocommerce/woocommerce-ios/pull/10652/files#r1319410614) shouldn't be needed on the latest `trunk`

File: Networking/Networking/Model/Order.swift [hunk 2]
```diff
@@ -156,6 +159,7 @@ public struct Order: Decodable, GeneratedCopiable, GeneratedFakeable {
         let status = try container.decode(OrderStatusEnum.self, forKey: .status)
 
         let currency = try container.decode(String.self, forKey: .currency)
+        let currencySymbol = try container.decode(String.self, forKey: .currencySymbol)
```
Comment (jaclync) [inline Networking/Networking/Model/Order.swift:162]: nit: might be worth adding a line of test in `OrderMapperTests.test_Order_fields_are_properly_parsed`?

File: Yosemite/Yosemite/Model/Storage/Order+ReadOnlyConvertible.swift [hunk 3]
```diff
@@ -91,6 +91,7 @@ extension Storage.Order: ReadOnlyConvertible {
                      number: number ?? "",
                      status: OrderStatusEnum(rawValue: statusKey),
                      currency: currency ?? "",
+                     currencySymbol: "", // Not stored in the Storage Layer, only used in the notifications extension.currency: currency ?? "",
```
Comment (jaclync) [inline Yosemite/Yosemite/Model/Storage/Order+ReadOnlyConvertible.swift]: super nit:

```suggestion
                     currencySymbol: "", // Not stored in the Storage Layer, only used in the notifications extension.
```

**Review by jaclync — APPROVED** [review]
LGTM 🚢

**Review by jostnes — APPROVED** [review]
i was pinged as part of @woocommerce/mobile-ui-testing-squad because some UI Tests file was updated. changes look good, thank you for updating the mock files and keeping the mock responses updated for the tests! 👍

**Pull request merged**